modules/openapi-generator/src/main/resources/Java/libraries/native/ApiClient.mustache

```diff
@@ -148,8 +148,28 @@ public class ApiClient {
    * Ctor.
    */
   public ApiClient() {
-    builder = HttpClient.newBuilder();
-    mapper = new ObjectMapper();
+    this.builder = createDefaultHttpClientBuilder();
```

Comment:
Soft: I liked the previous version better where this ctor invoked the other ctor `this(...)` but maybe you have a good reason.

Reply:
@MosheElisha 
https://cloud.drone.io/OpenAPITools/openapi-generator/14312/1/7

Reply:
:-) Oh, yes. This is why in one of my suggestions I moved that code to the other ctor:

```
  public ApiClient() { // Existing ctor refactored
    this(null, null, null); // Using new ctor
  }
  
  public ApiClient(Builder builder, ObjectMapper mapper, String baseUri) { // New ctor
    this.builder = builder != null ? builder : createDefaultHttpClientBuilder();
    this.mapper = mapper != null ? mapper : createDefaultObjectMapper();
    updateBaseUri(baseUri != null ? baseUri : getDefaultBaseUri());
    ...
```

but not sure this is nicer :-) Your call. Thank you!

Reply:
IMHO: this is a bad idea because it's allow user to do something line `new ApiClient(customHttpBuilder(), null, null)` which is not cool :)

this is why I prefer to either use default constructor or use fully customized constructor

Reply:
I personally don't see a problem with that but I don't have a strong opinion.

The disadvantage of the current approach is that if someone want to make a change in the ctor, it needs to be done in two places. This can be improved by moving the additional lines

```
interceptor = null;
readTimeout = null;
responseInterceptor = null;
```

into an `init()` method and call this method from both ctors.

**Actually, these lines can be deleted all together I believe because the default value is anyway `null`.** The `init()` method can be created in the future if there is a need.

Anyway, the current code is completely fine by me. Thank you!